ir/base.def

```diff
@@ -253,4 +254,20 @@ interface IInstance {
     virtual Type getType() const = 0;
 }
 
+/// An argument to a function call (or constructor call)
+/// Arguments may have optional names
+class Argument {
+    /// If an argument has no name the name.name is nullptr.
+    optional ID name;
+    Expression expression;
```

Comment:
Add

    Argument { if (!srcInfo && expression) srcInfo = expression->srcInfo; }

to always initialize the srcInfo of the argument from the expression if its not otherwise overridden, and all the places you explicitly copy the srcInfo from the expression will no longer need to do that.  It will also deal with any spot you missed -- much less error prone.

Reply:
I actually wanted to do this, but I wasn't sure about the syntax for constructors in the ir-generator...

Reply:
The syntax above (class name with no argument list) is an ir-generator special for a chunk of code to be added to all to auto-generated constructors it generates.  You can also declare a constructor normally (which will declare that constructor, overriding any auto-generated ctor that would have had the same signature).